doc-src/source/data-structure.rst

```diff
@@ -112,6 +113,135 @@ use the `rotate2` method::
+reference frame, and the columns are the unit vectors of the ENU
+coordinate system in the XYZ reference frame. That is, for this kind
+of simple rotation matrix between two orthogonal coordinate systems,
+the inverse rotation matrix is simply the transpose:
+
+.. math:: v_e = R^T \cdot v_i
+
+Heading, Pitch, Roll
+....................
+
+The instrument's *heading*, *pitch*, and *roll* information is not
+included in the data-structure by default. Instead, these variables
+can be computed from the orientation matrix using the
+``dolfyn.orient2euler`` function (``dolfyn.euler2orient`` provides the
+reverse functionality). This function computes these variables
+according to the following conventions:
+
+  - a "ZYX" rotation order. That is, these variables are computed
+    assuming that rotation from the earth -> instrument frame happens
+    by rotating around the z-axis first (heading), then rotating
+    around the y-axis (pitch), then rotating around the x-axis (roll).
+
+  - heading is defined as the direction the x-axis points, positive
+    clockwise from North (this is *opposite* the right-hand-rule
+    around the Z-axis)
+
+  - pitch is positive when the x-axis pitches up (this is *opposite* the
+    right-hand-rule around the Y-axis)
+
+  - roll is positive according to the right-hand-rule around the
+    instument's x-axis
+
+Considerable care has been taken to make sure
+that these definitions of *heading*, *pitch*, *roll* and *orientmat*
+are consistent within |dlfn| between instrument models.
+However, because the instrument manufacturer's definitions of these
+variables are not consistent between instrument makes/models, this
+means that |dlfn|\ 's consistent definitions are often different from
+the definitions provided by an instrument manufacturer (i.e., there is
+no consensus on these definitions, so |dlfn| uses the above
+definitions).
+
+So, while |dlfn| uses the instrument manufacturer's definition of the
+instrument coordinate system ("XYZ"), the details of how this relates
+to the 'earth' coordinate system, and how *pitch*, *roll*, *heading*
+are computed are often distinct from the definitions specified by
+the instrument manufacturer (e.g., some manufacturers reference
+heading off of the y-axis rather than x, some use a 'north-east-down'
+earth reference frame, etc.). For practical purposes, when utilizing
+|dlfn| orientation data (for all instrument types), this all means
+that the user should:
+
+  - Use the instrument manufacturers definitions of XYZ
+
+  - Interpret *heading*, *pitch*, and *roll* data according to the above
+    definitions (ignore manufacturer definitions of these variables)
+
+  - All rotations into the earth frame will yield vectors that are in
+    a ENU coordinate system
+
+It is also possible to obtain the raw *heading*, *pitch*, *roll* data
+as defined by the instrument manufacturer using the
+``keep_orient_raw`` argument in the :func:`dolfyn.read` function. That
```

Comment:
Is this displaying like you want it to?
:func:`dolfyn.read` function

Reply:
Will fix if needed in my heading-bug-fix-mcf2 branch. Expect pull request in future.